### PR TITLE
fix(cli/console): quote object symbol keys that are invalid identifiers

### LINKED ISSUE #7551: Inspect could be better for symbols
Here's a look at the current handling of `Symbol`s in `Deno.inspect`.

**1.** Neither Node nor Deno quote symbol names with unprintable characters, but both do escaping in object keys:

Deno

```console
> ({ [Symbol("foo\n\0bar")]: 1 })
{ Symbol(foo\n\x00bar): 1 }
```

Node

```console
> ({ [Symbol("foo\n\0bar")]: 1 })
{ [Symbol(foo\n\u0000bar)]: 1 }
```

**2.** However, note that Node puts `[]` around symbol keys, which is clearer in this contrived example:

Deno

```console
> ({ [Symbol()]: 1, 'Symbol()': 2 })
{ Symbol(): 2, Symbol(): 1 }
```

Node

```console
> ({ [Symbol()]: 1, 'Symbol()': 2 })
{ 'Symbol()': 2, [Symbol()]: 1 }
```

**3.** Lastly, while inspecting a symbol *directly*, neither Node nor Deno even do escaping (output is the same for both):

```console
> Symbol("foo\n\0bar")
Symbol(foo
bar)
```

### To do
  * [ ] 1. It seems okay not to quote symbol names, but maybe we should?
  * [ ] 2. I think we should add `[]` for object symbol keys like Node.
  * [ ] 3. When inspecting symbols directly we should at least do escaping — make it identical to the output for symbols in object keys (aside from the `[]`).

cc @caspervonb

### PR DESCRIPTION
**Pull request by caspervonb**
Partially resolves #7551

### CODE REVIEW

File: cli/rt/02_console.js [hunk 1]
```diff
@@ -722,7 +734,7 @@
     }
     for (const key of symbolKeys) {
       entries.push(
-        `${replaceEscapeSequences(key.toString())}: ${
+        `${maybeQuoteSymbol(key)}: ${
```
Comment (caspervonb) [inline cli/rt/02_console.js:737]: This needs tests.

**Comment by bartlomieju** [comment]
CC @nayeemrmn

File: cli/rt/02_console.js [hunk 2]
```diff
@@ -481,6 +481,19 @@
     return quoteString(string);
   }
 
+  // Surround a symbol's description in quotes when it is required (e.g the description has non printable characters).
+  function maybeQuoteSymbol(symbol) {
+    if (symbol.description === undefined) {
+      return symbol.toString();
+    }
+
+    if (/^[a-zA-Z_][a-zA-Z_.0-9]*$/.test(symbol.description)) {
```
Comment (nayeemrmn) [inline cli/rt/02_console.js:490]:
```suggestion
    if (/^[$a-zA-Z_][$a-zA-Z_.0-9]*$/.test(symbol.description)) {
```

**Review by nayeemrmn — COMMENTED** [review]
This isn't the behaviour of browsers and Node, but the Chrome behaviour doesn't make sense looking the way the following are formatted:
```
> console.log(["abc\ndef"]);
["abc↵def"]
> console.log([Symbol("abc\ndef")]);
[Symbol(abc
def)]
```
But maybe we should just always quote it instead of just for invalid identifiers.

File: cli/rt/02_console.js (hunk 2 above)
Comment (caspervonb) [reply cli/rt/02_console.js:490]: Fine with changing it but just want to point out that I deliberately omitted this from here and from key strings to match Node's behaviour (in respect to key strings, Symbol printing is wacky across the board).

**Review by ry — APPROVED** [review]
LGTM - thanks Casper

**Pull request merged**